src/ddmd/mars.d

```diff
@@ -1343,6 +1397,22 @@ Language changes listed by -transition=id:
             }
         }
     }
+    if (global.params.readStdin)
```

Comment:
Seems like this code should go into `root/file.d`, as reading from stdin should be a low level thing, not a high level one.